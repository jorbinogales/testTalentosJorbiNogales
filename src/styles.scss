/* You can add global styles to this file, and also import other style files */


/* Importing Bootstrap SCSS file. */
@import "~bootstrap/scss/bootstrap";

/* Importing Datepicker SCSS file. */
@import "~ngx-bootstrap/datepicker/bs-datepicker";

@font-face {
    font-family: 'monserrat';
    src: url('assets/lib/fonts/monserat/Montserrat-Medium.ttf');
}


@font-face {
    font-family: 'monserrat-bold';
    src: url('assets/lib/fonts/monserat/Montserrat-Bold.ttf');
}

@font-face {
    font-family: 'questrial';
    src: url('assets/lib/fonts/questrial/Questrial-Regular.ttf');
}



:root{
    --bg-pink: #ff6666;
    --bg-black: #666666;
    --bg-white: #cccccc;
}

button{
    color: var(--bg-pink);
    border-radius: 0px !important;
    font-size: 13px;
    font-family: 'questrial';
    &.btn-pink{
        background-color: var(--bg-pink) !important;
        color: white !important;
    }
}

.nav-item{
    display: block;
    padding: 2px 25px;
    &:hover{
        a{
            color: var(--bg-pink) !important;
        }
    }
    &.active{
        background-color: var(--bg-pink);
        a{
            color: white !important;
        }
    }
}


.wallpapper{
    position: relative;
    background-color: var(--bg-pink);
    display: block;
    margin:auto;
    height: 100%;
    width: 100%;
    min-height: 100vh;
    h2,p{
        position:relative;
        color: white;
        text-align: center;
    }
    p{
        font-size: 16px;
        font-family: 'questrial';
        top:250px;
        letter-spacing: 200;
    }
    h2{
        font-size: 39px;
        font-family: 'monserrat-bold';
        top:200px;
        letter-spacing: 300;
    }
    button{
        position: relative;
        max-width: 200px;
        margin:auto;
        display: block;
        top:270px;
        font-family: 'questrial';
        background: white;
        color: var(--bg-pink) !important;
        letter-spacing: 200;
    }
}

.menu{
    display: block;
    max-width: 700px;
    margin:auto;
    margin-top:10px;
    ul{ 
        display: flex !important;
        flex-wrap: wrap !important;
        flex-direction: inherit !important;
    }
    li{
        display: initial;
        margin:auto;
        cursor: pointer;
        a{
            color: var(--bg-black) !important;
            font-size: 14px;
            font-family: 'questrial';
        }
    }
}

.footer{
    padding: 25px 15px;
    p{
        color: var(--bg-white);
        text-align: center;
        font-family: 'questrial';
        font-size: 13px;
        letter-spacing: 5px;
        span{
            color: var(--bg-pink)
        }
    }
    ul{
        display: flex;
        flex-wrap: wrap;
        max-width: 300px;
        margin:auto;
        text-align: center;
        color: var(--bg-white);
        list-style-type: none;
        li{
            margin:auto;
            text-align: center;
            padding: 5px 10px;
        }
    }
}

.card{
    .card-img-overlay{
        opacity: 0;
    }
}

.card:hover{
    background-color: var(--bg-pink);
    cursor:pointer;
    img{
        opacity: .3;
        transition: all .5s ease-in-out;
    }
    h5{
        font-size: 18px;
        font-family: 'monserrat';
        color:white;
        display:block;
        margin-top:50px;
        text-align: center;
    }
    .card-img-overlay{
        opacity: 1;
    }
    .linea{
        display: block;
        height:1px;
        background-color: white;
        width: 80%;
        margin:auto;
        margin-top:20px;
    }
    .type{
        color:white;
        display: block;
        margin:auto;
        text-align: center;
        font-size: 18px;
        margin-top:50px;
        font-family: 'monserrat';
    }
}