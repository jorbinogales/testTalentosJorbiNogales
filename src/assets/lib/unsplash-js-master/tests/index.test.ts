import { CompleteRequestParams } from '../src/helpers/request';
import * as collections from '../src/methods/collections';
import * as photos from '../src/methods/photos';
import * as search from '../src/methods/search';
import * as users from '../src/methods/users';
import * as topics from '../src/methods/topics';
import { createApi } from '../src';
import { OrderBy } from '../src/types/request';
import { Language } from '../src/methods/search/types/request';
import { buildUrl, parseQueryAndPathname } from '../src/helpers/url';

describe('parseQueryAndPathname', () => {
  it('parses full URL correctly', () => {
    const output = parseQueryAndPathname(
      'https://api.unsplash.com/photos/foo123/download?foo=bar&baz=123#some-hash',
    );
    expect(output).toMatchSnapshot();
  });
  it('handles no hash correctly', () => {
    const output = parseQueryAndPathname(
      'https://api.unsplash.com/photos/foo123/download?foo=bar&baz=123',
    );
    expect(output).toMatchSnapshot();
  });

  it('handles hash without query correctly', () => {
    const output = parseQueryAndPathname(
      'https://api.unsplash.com/photos/foo123/download#some-hash',
    );
    expect(output).toMatchSnapshot();
  });

  it('handles no hash or query correctly', () => {
    const output = parseQueryAndPathname('https://api.unsplash.com/photos/foo123/download');
    expect(output).toMatchSnapshot();
  });

  it('handles query with only undefined properties correctly', () => {
    const output = parseQueryAndPathname('https://api.unsplash.com');
    expect(output).toMatchSnapshot();
  });
});

describe('buildUrl', () => {
  it('works with apiUrl (without path)', () => {
    const output = buildUrl({ pathname: '/foo/bar', query: { a: 1, b: 2 } })('https://example.com');
    expect(output).toEqual('https://example.com/foo/bar?a=1&b=2');
  });

  it('works with apiUrl (with path)', () => {
    const output = buildUrl({ pathname: '/foo/bar', query: { a: 1, b: 2 } })(
      'https://example.com/baz',
    );
    expect(output).toEqual('https://example.com/baz/foo/bar?a=1&b=2');
  });

  it('handles empty query correctly', () => {
    const output = buildUrl({ pathname: '/foo/bar', query: {} })('https://example.com');
    expect(output).toEqual('https://example.com/foo/bar');
  });
});

const PHOTO_ID = 'abc123';
const USERNAME = 'usernametest';
const SEARCH_QUERY = 'cat';
const COLLECTION_ID = 'collection123';
const TOPIC_ID_OR_SLUG = 'topic123';

type Api = ReturnType<typeof createApi>;
type Section = keyof Api;
const paramsTests: { [S in Section]: Record<keyof Api[S], CompleteRequestParams[]> } = {
  photos: {
    get: [photos.get.handleRequest({ photoId: PHOTO_ID })],
    list: [
      photos.list.handleRequest({ orderBy: OrderBy.LATEST, page: 4, perPage: 10 }),
      photos.list.handleRequest({ orderBy: OrderBy.LATEST }),
      photos.list.handleRequest({}),
    ],
    getStats: [photos.getStats.handleRequest({ photoId: PHOTO_ID })],
    getRandom: [
      photos.getRandom.handleRequest({}),
      photos.getRandom.handleRequest({
        collectionIds: ['a', 'bcd'],
        featured: true,
        orientation: 'landscape',
        username: USERNAME,
        query: SEARCH_QUERY,
        count: 1,
      }),
    ],
    trackDownload: [
      photos.trackDownload.handleRequest({
        downloadLocation: 'https://api.unsplash.com/photos/foo123/download',
      }),
      photos.trackDownload.handleRequest({
        downloadLocation:
          'https://api.unsplash.com/photos/foo123/download?foo=bar&baz=123#some-hash',
      }),
    ],
  },
  users: {
    getPhotos: [
      users.getPhotos.handleRequest({ username: USERNAME }),
      users.getPhotos.handleRequest({ username: USERNAME, stats: true, page: 4, perPage: 10 }),
    ],
    getCollections: [users.getCollections.handleRequest({ username: USERNAME })],
    getLikes: [users.getLikes.handleRequest({ username: USERNAME })],
    get: [users.get.handleRequest({ username: USERNAME })],
  },
  search: {
    getCollections: [search.getCollections.handleRequest({ query: SEARCH_QUERY })],
    getPhotos: [
      search.getPhotos.handleRequest({ query: SEARCH_QUERY }),
      search.getPhotos.handleRequest({
        query: SEARCH_QUERY,
        collectionIds: ['a', 'b', 'c'],
        color: 'blue',
        contentFilter: 'high',
        lang: Language.Esperanto,
        orientation: 'portrait',
        orderBy: 'latest',
        page: 123,
        perPage: 4,
      }),
    ],
    getUsers: [search.getUsers.handleRequest({ query: SEARCH_QUERY })],
  },
  collections: {
    getPhotos: [
      collections.getPhotos.handleRequest({ collectionId: COLLECTION_ID }),
      collections.getPhotos.handleRequest({
        collectionId: COLLECTION_ID,
        orientation: 'landscape',
      }),
    ],
    get: [collections.get.handleRequest({ collectionId: COLLECTION_ID })],
    list: [
      collections.list.handleRequest({}),
      collections.list.handleRequest({ page: 8, perPage: 23 }),
    ],
    getRelated: [collections.getRelated.handleRequest({ collectionId: COLLECTION_ID })],
  },
  topics: {
    get: [topics.get.handleRequest({ topicIdOrSlug: TOPIC_ID_OR_SLUG })],
    getPhotos: [
      topics.getPhotos.handleRequest({
        topicIdOrSlug: TOPIC_ID_OR_SLUG,
      }),
      topics.getPhotos.handleRequest({
        topicIdOrSlug: TOPIC_ID_OR_SLUG,
        orientation: 'portrait',
        page: 123,
        perPage: 4,
        orderBy: OrderBy.LATEST,
      }),
    ],
    list: [
      topics.list.handleRequest({}),
      topics.list.handleRequest({
        orderBy: 'position',
        topicIdsOrSlugs: [TOPIC_ID_OR_SLUG],
        page: 2,
        perPage: 13,
      }),
    ],
  },
};

describe('requestParams', () => {
  Object.entries(paramsTests).forEach(([sectionName, section]) => {
    describe(`${sectionName}`, () => {
      Object.entries(section).forEach(([methodName, methodTests]) => {
        describe(`${methodName}: `, () => {
          it('Correcty builds request arguments', () => {
            methodTests.forEach(output => expect(output).toMatchSnapshot());
          });
        });
      });
    });
  });
  describe('additional arguments', () => {
    it('Correctly passes down additional args', () => {
      const abortcontroller = new AbortController();
      const signal = abortcontroller.signal;
      const output = photos.getRandom.handleRequest({}, { signal });

      expect(output).toMatchSnapshot();
    });
    it('Correctly merges headers', () => {
      const output = photos.getRandom.handleRequest({}, { headers: { 'test-headers': 'foo' } });

      expect(output).toMatchSnapshot();
    });
  });
});
