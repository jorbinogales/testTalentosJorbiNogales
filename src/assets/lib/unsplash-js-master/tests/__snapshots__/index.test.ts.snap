// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseQueryAndPathname handles hash without query correctly 1`] = `
Object {
  "pathname": "/photos/foo123/download",
  "query": Object {},
}
`;

exports[`parseQueryAndPathname handles no hash correctly 1`] = `
Object {
  "pathname": "/photos/foo123/download",
  "query": Object {
    "baz": "123",
    "foo": "bar",
  },
}
`;

exports[`parseQueryAndPathname handles no hash or query correctly 1`] = `
Object {
  "pathname": "/photos/foo123/download",
  "query": Object {},
}
`;

exports[`parseQueryAndPathname handles query with only undefined properties correctly 1`] = `
Object {
  "pathname": undefined,
  "query": Object {},
}
`;

exports[`parseQueryAndPathname parses full URL correctly 1`] = `
Object {
  "pathname": "/photos/foo123/download",
  "query": Object {
    "baz": "123",
    "foo": "bar",
  },
}
`;

exports[`requestParams additional arguments Correctly merges headers 1`] = `
Object {
  "headers": Object {
    "cache-control": "no-cache",
    "test-headers": "foo",
  },
  "pathname": "/photos/random",
  "query": Object {},
}
`;

exports[`requestParams additional arguments Correctly passes down additional args 1`] = `
Object {
  "headers": Object {
    "cache-control": "no-cache",
  },
  "pathname": "/photos/random",
  "query": Object {},
  "signal": AbortSignal {},
}
`;

exports[`requestParams collections get:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/collections/collection123",
  "query": Object {},
}
`;

exports[`requestParams collections getPhotos:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/collections/collection123/photos",
  "query": Object {},
}
`;

exports[`requestParams collections getPhotos:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/collections/collection123/photos",
  "query": Object {
    "orientation": "landscape",
  },
}
`;

exports[`requestParams collections getRelated:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/collections/collection123/related",
  "query": Object {},
}
`;

exports[`requestParams collections list:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/collections",
  "query": Object {},
}
`;

exports[`requestParams collections list:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/collections",
  "query": Object {
    "page": 8,
    "per_page": 23,
  },
}
`;

exports[`requestParams photos get:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/photos/abc123",
  "query": Object {},
}
`;

exports[`requestParams photos getRandom:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {
    "cache-control": "no-cache",
  },
  "pathname": "/photos/random",
  "query": Object {},
}
`;

exports[`requestParams photos getRandom:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {
    "cache-control": "no-cache",
  },
  "pathname": "/photos/random",
  "query": Object {
    "collections": "a,bcd",
    "count": 1,
    "featured": true,
    "orientation": "landscape",
    "query": "cat",
    "username": "usernametest",
  },
}
`;

exports[`requestParams photos getStats:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/photos/abc123/statistics",
  "query": Object {},
}
`;

exports[`requestParams photos list:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/photos",
  "query": Object {
    "order_by": "latest",
    "page": 4,
    "per_page": 10,
  },
}
`;

exports[`requestParams photos list:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/photos",
  "query": Object {
    "order_by": "latest",
  },
}
`;

exports[`requestParams photos list:  Correcty builds request arguments 3`] = `
Object {
  "headers": Object {},
  "pathname": "/photos",
  "query": Object {},
}
`;

exports[`requestParams photos trackDownload:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/photos/foo123/download",
  "query": Object {},
}
`;

exports[`requestParams photos trackDownload:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/photos/foo123/download",
  "query": Object {
    "baz": "123",
    "foo": "bar",
  },
}
`;

exports[`requestParams search getCollections:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/search/collections",
  "query": Object {
    "query": "cat",
  },
}
`;

exports[`requestParams search getPhotos:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/search/photos",
  "query": Object {
    "query": "cat",
  },
}
`;

exports[`requestParams search getPhotos:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/search/photos",
  "query": Object {
    "collections": "a,b,c",
    "color": "blue",
    "content_filter": "high",
    "lang": "eo",
    "order_by": "latest",
    "orientation": "portrait",
    "page": 123,
    "per_page": 4,
    "query": "cat",
  },
}
`;

exports[`requestParams search getUsers:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/search/users",
  "query": Object {
    "query": "cat",
  },
}
`;

exports[`requestParams topics get:  Correcty builds request arguments 1`] = `
Object {
  "pathname": "/topics/topic123",
  "query": Object {},
}
`;

exports[`requestParams topics getPhotos:  Correcty builds request arguments 1`] = `
Object {
  "pathname": "/topics/topic123/photos",
  "query": Object {},
}
`;

exports[`requestParams topics getPhotos:  Correcty builds request arguments 2`] = `
Object {
  "pathname": "/topics/topic123/photos",
  "query": Object {
    "order_by": "latest",
    "orientation": "portrait",
    "page": 123,
    "per_page": 4,
  },
}
`;

exports[`requestParams topics list:  Correcty builds request arguments 1`] = `
Object {
  "pathname": "/topics",
  "query": Object {},
}
`;

exports[`requestParams topics list:  Correcty builds request arguments 2`] = `
Object {
  "pathname": "/topics",
  "query": Object {
    "ids": "topic123",
    "order_by": "position",
    "page": 2,
    "per_page": 13,
  },
}
`;

exports[`requestParams users get:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/users/usernametest",
  "query": Object {},
}
`;

exports[`requestParams users getCollections:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/users/usernametest/collections",
  "query": Object {},
}
`;

exports[`requestParams users getLikes:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/users/usernametest/likes",
  "query": Object {},
}
`;

exports[`requestParams users getPhotos:  Correcty builds request arguments 1`] = `
Object {
  "headers": Object {},
  "pathname": "/users/usernametest/photos",
  "query": Object {},
}
`;

exports[`requestParams users getPhotos:  Correcty builds request arguments 2`] = `
Object {
  "headers": Object {},
  "pathname": "/users/usernametest/photos",
  "query": Object {
    "page": 4,
    "per_page": 10,
    "stats": true,
  },
}
`;
